src/app/daos/[id]/page.tsx

```diff
@@ -1,32 +1,96 @@
 "use client";
 
 import { useParams } from "next/navigation";
+import { useQuery } from "@tanstack/react-query";
 import DAOOverview from "@/components/daos/dao-overview";
-import { useDAODetails } from "@/hooks/use-dao-details";
 import { Skeleton } from "@/components/ui/skeleton";
+import {
+  fetchDAO,
+  fetchToken,
+  fetchDAOExtensions,
+  fetchMarketStats,
+  fetchTreasuryTokens,
+  fetchTokenPrice,
+} from "@/queries/daoQueries";
 
 export const runtime = "edge";
 
 export default function DAOPage() {
   const params = useParams();
   const id = params.id as string;
-  const { dao, token, loading, marketStats, treasuryTokens } =
-    useDAODetails(id);
 
-  if (loading) {
+  const { data: dao, isLoading: isLoadingDAO } = useQuery({
+    queryKey: ["dao", id],
+    queryFn: () => fetchDAO(id),
+  });
+
+  const { data: token, isLoading: isLoadingToken } = useQuery({
+    queryKey: ["token", id],
+    queryFn: () => fetchToken(id),
+    enabled: !!dao,
+  });
+
+  const { data: extensions, isLoading: isLoadingExtensions } = useQuery({
+    queryKey: ["extensions", id],
+    queryFn: () => fetchDAOExtensions(id),
+    enabled: !!dao,
+  });
+
+  const dex = extensions?.find((ext) => ext.type === "dex")?.contract_principal;
+  const treasuryAddress = extensions?.find(
+    (ext) => ext.type === "aibtc-treasury"
+  )?.contract_principal;
+
+  const { data: tokenPrice, isLoading: isLoadingTokenPrice } = useQuery({
+    queryKey: ["tokenPrice", dex],
+    queryFn: () => fetchTokenPrice(dex!),
+    enabled: !!dex,
+  });
+
+  const { data: marketStats, isLoading: isLoadingMarketStats } = useQuery({
+    queryKey: [
+      "marketStats",
+      id,
+      dex,
+      token?.contract_principal,
+      token?.symbol,
+      token?.max_supply,
+    ],
+    queryFn: () =>
+      fetchMarketStats(
+        dex!,
+        token!.contract_principal,
+        token!.symbol,
+        token!.max_supply || 0
+      ),
+    enabled: !!dex && !!token && !!token.contract_principal && !!token.symbol,
+  });
+
+  const { data: treasuryTokens, isLoading: isLoadingTreasuryTokens } = useQuery(
+    {
+      queryKey: ["treasuryTokens", treasuryAddress, tokenPrice?.price],
+      queryFn: () => fetchTreasuryTokens(treasuryAddress!, tokenPrice!.price),
+      enabled: !!treasuryAddress && !!tokenPrice,
+    }
+  );
+
+  const isLoading =
+    isLoadingDAO ||
+    isLoadingToken ||
+    isLoadingExtensions ||
+    isLoadingTokenPrice ||
+    isLoadingMarketStats ||
+    isLoadingTreasuryTokens;
+
+  if (isLoading) {
```

Comment:
Making a note that we probably want to break the loading out into skeletons and/or separate sections where possible.

I'd rather see 90% of my layout and a few spots loading values instead of getting stuck behind a spinner waiting for the whole page. Avoiding layout shift is critical, though.